service/rest.go

```diff
@@ -0,0 +1,99 @@
+package service
+
+import (
+	"net/http"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/auth"
+	"github.com/evergreen-ci/evergreen/model"
+	"github.com/evergreen-ci/gimlet"
+	"github.com/pkg/errors"
+	"github.com/urfave/negroni"
+)
+
+// restContextKey is the type used to store
+type restContextKey int
+
+const RestContext restContextKey = 0
+
+type restAPIService interface {
+	GetSettings() evergreen.Settings
+	LoggedError(http.ResponseWriter, *http.Request, int, error)
+}
+
+type restAPI struct {
+	restAPIService
+}
+
+type restV1middleware struct {
+	restAPIService
+}
+
+func (ra *restV1middleware) ServeHTTP(rw http.ResponseWriter, r *http.Request, next http.HandlerFunc) {
+	vars := gimlet.GetVars(r)
+	ctx, err := model.LoadContext(vars["task_id"], vars["build_id"], vars["version_id"], vars["patch_id"], vars["project_id"])
```

Comment:
project can also be stored in a cookie rather than the url (see the old getRequestProjectId func)

Reply:
as discussed in person, this is only true of the UI server, which has a very similar method.